akka-remote/src/main/scala/akka/remote/Remoting.scala

```diff
@@ -549,6 +549,25 @@ private[remote] class EndpointManager(conf: Config, log: LoggingAdapter) extends
         case (Some((endpoint, currentUid)), Some(quarantineUid)) if currentUid == quarantineUid ⇒ context.stop(endpoint)
         case _ ⇒ // nothing to stop
       }
+
+      // Stop all stashed pending connections
+      pendingReadHandoffs = pendingReadHandoffs.filter {
+        case (pendingActor, pendingHandle) ⇒
+          val addressMatches = pendingHandle.remoteAddress == address
+          // This yields true also if the Quarantine command did not contain a UID
+          val uidMatches = uidToQuarantineOption.forall(_ == pendingHandle.handshakeInfo.uid)
+          val keep = addressMatches & uidMatches
+
+          // Side-effecting here
+          if (!keep) {
+            if (uidToQuarantineOption.nonEmpty)
+              pendingHandle.disassociate(AssociationHandle.Quarantined)
+            else pendingHandle.disassociate()
+            context.stop(pendingActor)
+          }
+          keep
```

Comment:
now we keep the ones that match the quarantine and throw away the others, is that intended?